tests/test_virtual_chassis.py

```diff
@@ -224,7 +256,145 @@ def test_chassis_system_neigh(self, vct):
                 assert encap_index == sys_neigh_encap_index, "Encap index not sync-ed correctly"
                 
                 break
+        
+    def test_chassis_system_lag(self, vct):
+        """Test portchannel in VOQ based chassis systems.
+        
+        This test validates that
+           (i)   PortChannel is created in local asic with system port aggregator id (system lag id)
+                     - Unique lag id is allocated from chassis app db in supervisor card
+                     - The unique lag id is sent in system port aggregator id attribute
+           (ii)  PortChannel members are successfully added in the PortChannel created 
+           (iii) Local PortChannel is synced in chassis app db
```

Comment:
Can we add few negative tests to cover the failure conditions to cover the logic in lua id generation either here .. or in the other PR https://github.com/Azure/sonic-swss/pull/1603

Reply:
Yes. I'll add test cases for table full error. For delete and get error, it will be difficult to drive the tests via app db. The execution will not go that for to invoke the lagid allocator apis. I'll try to add test cases for these by invoking lua directly from cli using redis eval command, if possible

Reply:
Fixed. Added test cases to test lag id allocator table full error,  lag id delete error, re-use of released lag id.